DiscountDiscover/View Controllers/SignUpViewController.m

```diff
@@ -0,0 +1,75 @@
+//
+//  SignUpViewController.m
+//  DiscountDiscover
+//
+//  Created by Kaitlyn Gunadhi on 7/23/20.
+//  Copyright © 2020 Kaitlyn Gunadhi. All rights reserved.
+//
+
+#import "SignUpViewController.h"
+#import <Parse/Parse.h>
+#import "LoginViewController.h"
+#import "UIViewController+Error.h"
+
+@interface SignUpViewController ()
+
+@property (weak, nonatomic) IBOutlet UITextField *emailField;
+@property (weak, nonatomic) IBOutlet UITextField *nameField;
+@property (weak, nonatomic) IBOutlet UITextField *usernameField;
+@property (weak, nonatomic) IBOutlet UITextField *passwordField;
+@property (weak, nonatomic) IBOutlet UIButton *signupButton;
+
+@end
+
+@implementation SignUpViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.signupButton.layer.cornerRadius = 7;
+    self.signupButton.clipsToBounds = YES;
+}
+
+- (IBAction)onTap:(id)sender {
+    [self.view endEditing:YES];
+}
+
+- (IBAction)registerUser:(id)sender {
+    // initialize a user object
+    PFUser *newUser = [PFUser user];
+    
+    // set user properties
+    newUser.email = self.emailField.text;
+    newUser[@"name"] = self.nameField.text;
+    newUser.username = self.usernameField.text;
+    newUser.password = self.passwordField.text;
+    
+    if ([self emptyField]) {
+        [self showErrorAlert:@"Invalid Entry" message:@"Please complete all fields."];
+    } else {
+        // call sign up function on the user
+        __weak SignUpViewController *weakSelf = self;
+        [newUser signUpInBackgroundWithBlock:^(BOOL succeeded, NSError * error) {
+            SignUpViewController *strongSelf = weakSelf;
+            if (error != nil) {
+                [strongSelf showErrorAlert:@"Sign Up Error" message:error.localizedDescription];
```

Comment:
Does the Parse documentation state that these blocks are called on the main thread? If not it would be safer to explicitly wrap any code that needs to run on the main thread in a `dispatch_async(dispatch_get_main_queue(), ^{...});` call to be safe. Even if it works now, it might change without notice in future releases of the Parse SDK if it's not documented.